code/modules/vehicles/armored/som_armored_weapons.dm

```diff
@@ -14,27 +17,28 @@
 	///Range of this weapon
 	var/beam_range = 20
 	///Armor pen of this weapon
-	var/armor_pen = 30
+	var/armor_pen = 20
 
 /obj/item/armored_weapon/volkite_carronade/do_fire(turf/source_turf, ammo_override)
 	var/turf/target_turf = get_turf_in_angle(Get_Angle(source_turf, get_turf(current_target)), source_turf, beam_range)
-
 	var/list/turf/beam_turfs = get_line(source_turf, target_turf)
 	var/list/turf/impacted_turfs = list()
 	var/list/light_effects = list()
+	var/list/stop_beam_turfs
```

Comment:
I don't quite get what this var is doing/for. Is it so you hit tile behind the stopping point?

Reply:
so the final end point can change. Initially we just check for dense turfs, since we need to check through contents later.
If it later hits something that should stop the beam, thats when the loop checks if remaining turfs are in 1 range of the new stop turf (since it just ignores you if you remove them from the list mid loop).

Reply:
Would appreciate commenting to make that clearer, the whole thing is pretty rough for me to parse.

Reply:
updated